dubbo-spring-boot/dubbo-spring-boot-observability-starter/src/main/java/org/apache/dubbo/spring/boot/observability/autoconfigure/otel/OpenTelemetryAutoConfiguration.java

```diff
@@ -0,0 +1,257 @@
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * provider OpenTelemetry when you are using Boot <3.0 or you are not using spring-boot-starter-actuator
+ * @author shenfeng
+ */
+@AutoConfiguration(before = DubboMicrometerTracingAutoConfiguration.class, afterName = "org.springframework.boot.actuate.autoconfigure.tracing.OpenTelemetryAutoConfiguration")
+@ConditionalOnDubboTracingEnable
+@ConditionalOnClass({OtelTracer.class, SdkTracerProvider.class, OpenTelemetry.class})
+@EnableConfigurationProperties(DubboTracingProperties.class)
+public class OpenTelemetryAutoConfiguration {
+
+    /**
+     * Default value for application name if {@code spring.application.name} is not set.
+     */
+    private static final String DEFAULT_APPLICATION_NAME = "application";
+
+    private final DubboTracingProperties dubboTracingProperties;
+
+    OpenTelemetryAutoConfiguration(DubboTracingProperties dubboTracingProperties) {
+        this.dubboTracingProperties = dubboTracingProperties;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    OpenTelemetry openTelemetry(SdkTracerProvider sdkTracerProvider, ContextPropagators contextPropagators) {
+        return OpenTelemetrySdk.builder().setTracerProvider(sdkTracerProvider).setPropagators(contextPropagators)
+            .build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    SdkTracerProvider otelSdkTracerProvider(Environment environment, ObjectProvider<SpanProcessor> spanProcessors,
+                                            Sampler sampler) {
+        String applicationName = environment.getProperty("spring.application.name", DEFAULT_APPLICATION_NAME);
+        SdkTracerProviderBuilder builder = SdkTracerProvider.builder().setSampler(sampler)
+            .setResource(Resource.create(Attributes.of(ResourceAttributes.SERVICE_NAME, applicationName)));
+        spanProcessors.orderedStream().forEach(builder::addSpanProcessor);
+        return builder.build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    ContextPropagators otelContextPropagators(ObjectProvider<TextMapPropagator> textMapPropagators) {
+        return ContextPropagators.create(TextMapPropagator.composite(textMapPropagators.orderedStream().collect(Collectors.toList())));
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    Sampler otelSampler() {
+        Sampler rootSampler = Sampler.traceIdRatioBased(this.dubboTracingProperties.getSampling().getProbability());
+        return Sampler.parentBased(rootSampler);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    SpanProcessor otelSpanProcessor(ObjectProvider<SpanExporter> spanExporters,
```

Comment:
I see that there is no @ConditionalOnMissingBean in the latest springboot3: 
![image](https://user-images.githubusercontent.com/56248584/221387428-ad73cad3-ae16-4eef-9561-8287681da315.png)

Reply:
`OpenTelemetryAutoConfiguration` is after `org.springframework.boot.actuate.autoconfigure.tracing.OpenTelemetryAutoConfiguration`,so we need add  `@ConditionalOnMissingBean` use springboot3 default

Reply:
Okay, I see.